### DISCUSSION TITLE
Plans for subscription vs. non-subscription webhooks

### DISCUSSION THREAD

**Discussion by rrrnld** [saleor/saleor#12626] [Q&A] [answered]
I noticed that when clicking to configure new webhook events for an app in the dashboard, it automatically creates a subscription webhook with a customizable payload. We have several already-existing applications that use the predefined webhook payloads. Given that the interface nudges you to use the subscription webhooks as it does, are there any deprecation plans for the "normal" webhooks? I couldn't find anything about that in the documentation, and would be happy if you could clarify if we need to write new code and integration tests. Thanks!

**Comment by korycins** [comment, marked as answer]
Hey @heyarne 
In the future, we plan to fully transition to subscriptions. As a result, static payloads may be deprecated and removed, or simplified to provide only an `id` related to the object that triggered the event. We have not yet announced this as deprecated as we want to ensure that webhook subscriptions fully cover static payloads first. If you are building a new application, we suggest using webhook subscriptions.

Regarding your already-existing application, it sounds like there may be a bug. The dashboard should allow you to define webhooks with the old payload. Without this, you will not be able to install any custom app that was built with static webhooks.

**Reply by rrrnld** [reply]
Thanks for your clarification! We definitely value non-breaking updates. Commerce is not the core business of my current employer and we have found Saleor to be moving quite fast.
